src/Connector.js

```diff
@@ -37,17 +39,33 @@ class Connector {
   }
 
   async listenToCommands() {
-    await this.client.on('getData', async (cmd) => {
-      const { id, sensorIds } = cmd;
+    const { devices = [] } = await this.client.getDevices();
+    this.devices = devices
+      .filter((device) => !!device.schema)
+      .map((device) => device.id);
+
+    await Promise.all(
+      this.devices.map((device) => this.registerListeners(device))
+    );
+  }
+
+  async registerListeners(thingId) {
+    await this.client.on(`device.${thingId}.data.request`, async (msg) => {
+      const { id, sensorIds } = msg;
       this.onDataRequestedCb(id, sensorIds);
     });
 
-    await this.client.on('setData', async (cmd) => {
-      const { id, data } = cmd;
+    await this.client.on(`device.${thingId}.data.update`, async (msg) => {
+      const { id, data } = msg;
       this.onDataUpdatedCb(id, data);
     });
   }
 
+  async clearListeners(thingId) {
+    await this.client.clear(`device.${thingId}.data.request`);
```

Comment:
Shouldn't it be `this.client.unsubscribe(...)`?

Reply:
You're right! Now it's synced with SDK code

Reply:
Updated! :smiley: